### ISSUE TITLE
Exception with an error code: 0x24 (SPXERR_EXTENSION_LIBRARY_NOT_FOUND) on Linux-ARM64

### ISSUE THREAD

**Issue by CobraCalle** [Azure-Samples/cognitive-services-speech-sdk#587]
**Describe the bug**
When using keyword detection on linux arm 64 there is an error since version 1.11. 1.10 worked but had a reliability bug that is fixed with 1.11.

**To Reproduce**
1. Use keyword detection on linux arm64
2. Call StartKeywordDetection

**Expected behavior**
Does not throw an exception and works like on other platforms

**Version of the Cognitive Services Speech SDK**
1.11

**Platform, Operating System, and Programming Language**
 - OS: Linux (ubuntu)
 - Hardware - Raspberry Pi ARM64
 - Programming language: C#

**Additional context**
The 1.10 version hat a glitch too, because I had to rename libMicrosoft.CognitiveServices.Speech.core.so to libMicrosoft.CognitiveServices.Speech.core.dll.so because it tried to load the wrong file name. But after renaming the file it worked. With the new version (which works fine won windows) I can not get it to work...

**Comment by pankopon** [comment]
Hi, can you please try the attached simple test program for keyword detection and report the results? I ran it with the following setup and had no errors:
- Ubuntu 18.04.4 LTS arm64 on RPi 4
- dotnet version 3.1.200
- Speech SDK 1.11.0 and netcoreapp3.0 as configured in .csproj

[helloworld_keyword.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/4495926/helloworld_keyword.zip)

Extract the zip and edit Program.cs, replace "YourXyz" strings with your info (subscription key, region, file paths). The program reads input (keyword + some speech) from a wav file by default, but if you want to use microphone then replace

```
using (var audioInput = AudioConfig.FromWavFileInput(@"YourKeywordAndSpeechInputFile.wav"))
{
    using (var recognizer = new SpeechRecognizer(config, audioInput))
```

with

```
    using (var recognizer = new SpeechRecognizer(config))
```

(and remove the closing bracket for `using (var audioInput ...` at the end)

Next go to the directory helloworld_keyword and run commands

```
dotnet nuget locals all --clear
dotnet build helloworld/helloworld.csproj
dotnet helloworld/bin/Debug/netcoreapp3.0/helloworld.dll
```

The important part is to clear all files from all local NuGet cache directories first. Proper support for .NET Core C# on Linux ARM32/64 was only added in the Speech SDK 1.11.0 release (previously it was indeed incomplete), so it's better to start with a clean state.

**Comment by CobraCalle** [comment]
Hello,

thanks for your fast response.

I've run your test-program (but on ubuntu 19.1 and as a self container package, because thats my current setup, and I don´t think that taht will make any difference. I hope that would not make any difference).

Here is the result:
ubuntu@ubuntu:~/test$ ./helloworld
Unhandled exception. System.ApplicationException: Exception with an error code: 0x24 (SPXERR_EXTENSION_LIBRARY_NOT_FOUND)
   at Microsoft.CognitiveServices.Speech.Internal.SpxExceptionThrower.ThrowIfFail(IntPtr hr)
   at Microsoft.CognitiveServices.Speech.Recognizer.StartKeywordRecognition(KeywordRecognitionModel model)
   at Microsoft.CognitiveServices.Speech.SpeechRecognizer.<>c__DisplayClass46_0.<StartKeywordRecognitionAsync>b__1()
   at Microsoft.CognitiveServices.Speech.Recognizer.DoAsyncRecognitionAction(Action recoImplAction)
   at Microsoft.CognitiveServices.Speech.SpeechRecognizer.<>c__DisplayClass46_0.<StartKeywordRecognitionAsync>b__0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback,
 Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at helloworld.Program.ContinuousRecognitionWithKeywordSpottingAsync() in D:\Temp\helloworld_keyword\helloworld\Program.cs:line 91
   at helloworld.Program.Main() in D:\Temp\helloworld_keyword\helloworld\Program.cs:line 106
   at helloworld.Program.<Main>()
Aborted (core dumped)

**Comment by CobraCalle** [comment]
just to be sure... now I´ve installed dotnet core 3.0, build the test app on the device and started that version... same result

Is it possible that ubuntu 19 is really the problem?

ubuntu@ubuntu:~/helloworld_keyword$ dotnet helloworld/bin/Debug/netcoreapp3.0/helloworld.dll
Unhandled exception. System.ApplicationException: Exception with an error code: 0x24 (SPXERR_EXTENSION_LIBRARY_NOT_FOUND)
   at Microsoft.CognitiveServices.Speech.Internal.SpxExceptionThrower.ThrowIfFail(IntPtr hr)
   at Microsoft.CognitiveServices.Speech.Recognizer.StartKeywordRecognition(KeywordRecognitionModel model)
   at Microsoft.CognitiveServices.Speech.SpeechRecognizer.<>c__DisplayClass46_0.<StartKeywordRecognitionAsync>b__1()
   at Microsoft.CognitiveServices.Speech.Recognizer.DoAsyncRecognitionAction(Action recoImplAction)
   at Microsoft.CognitiveServices.Speech.SpeechRecognizer.<>c__DisplayClass46_0.<StartKeywordRecognitionAsync>b__0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback,
 Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback,
 Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at helloworld.Program.ContinuousRecognitionWithKeywordSpottingAsync() in /home/ubuntu/helloworld_keyword/helloworld/Program.cs:line 91
   at helloworld.Program.Main() in /home/ubuntu/helloworld_keyword/helloworld/Program.cs:line 106
   at helloworld.Program.<Main>()
Aborted (core dumped)

**Comment by pankopon** [comment]
I was actually now able to reproduce the error, but the cause is a bit strange. It seems that on ARM (both 32 and 64 bit) only the native core library (libMicrosoft.CognitiveServices.Speech.core.so) is found as expected, but the native extension libraries (libMicrosoft.CognitiveServices.Speech.extension.codec.so and libMicrosoft.CognitiveServices.Speech.extension.kws.so) are not even though they are located in the same directory.

To overcome this, it is necessary to set the shared library search path LD_LIBRARY_PATH to point at the location of native SDK libraries. For example, while in the helloworld_keyword directory, run a command in bash:

```
export LD_LIBRARY_PATH=$(pwd)/helloworld/bin/Debug/netcoreapp3.0/runtimes/linux-arm64/native
```

The reason why I didn't see the error at first was that my setup had LD_LIBRARY_PATH set by default, due to other (non-C#) SDK testing, and so the extension libraries were resolved without errors. Also our automated test environment has the same. We'll check if there's something to change in the SDK configuration to avoid the issue, as it seems ARM specific.

We only use Ubuntu LTS releases (16.04 and 18.04 at the moment) for testing on Ubuntu, so it is possible that some newer version might occasionally trigger issues, but that's probably not the case here.

**Comment by CobraCalle** [comment]
Thanks you very much... I can confiirm, setting the library path solves the problem.

**Comment by pankopon** [comment]
Just to update, this will be fixed in the 1.12.0 release so that it won't be necessary to manually set the library search path anymore.

**Comment by jhakulin** [comment]
Closing the issue as it will be fixed in 1.12.0 release coming May 2020.

**Comment by MarcoAndrade26** [comment]
How did you installed speech services on Raspberry pi?
